Assets/Scripts/Button/RunButton.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using UnityEngine;
+
+namespace DN
+{
+	/// <summary>
+	/// Uses event to start puzzle.
+	/// </summary>
+	public class RunButton : MonoBehaviour
+	{
+		public event Action RunPuzzleEvent;
+
+		public void RunPuzzle()
+		{
+			RunPuzzleEvent?.Invoke();
+		}
+	}
```

Comment:
Het is beter dat de button actions in dit script gebeuren, in plaats van dat de knop er voor zorgt. Dit zodat geen andere class `RunPuzzle` kan callen. Misschien ook handiger om het gewoon `Run` te noemen ipv `RunPuzzle` maar dat maakt niet echt uit.

Dus gewoon een Button variable toevoegen aan dit script, en dan `button.onClick.AddListener(RunPuzzle)` gebruiken om te subscriben. Vergeet dan ook niet `button.onClick.RemoveListener` te gebruiken, `button.onClick.RemoveAllListeners` mag ook.

Reply:
fixed